### DISCUSSION TITLE
Ghosts.exe...where is it?

### DISCUSSION THREAD

**Discussion by MoshFarm** [cmu-sei/GHOSTS#78] [Q&A] [answered]
I may be a bit of noob using GHOSTS, but I can't seem to locate the GHOSTS.exe file within the Ghosts.Client directory or subdirectories.  Am I missing something obvious? Trying to get the client running on a windows machine ,but can't find the .exe to start it. 

Thanks for any insight you can provide.

MF

**Comment by sei-dupdyke** [comment, marked as answer]
Hello, and thanks for checking out the project!

If you are following along with the [installation from binaries instructions](https://github.com/cmu-sei/GHOSTS/wiki/Installation-from-distribution-binaries), the Windows-specific installers should have .exe files within them. The linux distributions use a different binary structure and so they will contain a different file for execution.

**Reply by RANDW0LF** [reply]
I note the latest installation from binaries is v4, and that there is a beta v6 as well. Is not an installer for 5.1?
Do you have any good links on how to compile the source to create your own .exe setup?

**Reply by sei-dupdyke** [reply]
IIRC, parts of the framework made it to v5, but not others, and so I decided to realign things across the board on v6 with this next release.

Compiling should be relatively straightforward, I try hard to make the process as simple as download and build, but there are a few gotchas:

1. I'm using Visual Studio 2017 for the Windows client on Windows, and Jetbrains Rider on Mac for everything else. I don't think you'll have any issues building with other IDEs (or a text editor and the command line for that matter), but I've not tested this. To be clear, you have to build the Windows client on Windows, and the rest(API, Linux client) can be built on any OS.

2. You'll have to install the relevant versions of the dotnet SDK on your build machine (traditional dotnet 4.6.1 for the Windows client, dotnet 6 for everything else). 

As a side note, we've had the Windows client pinned at 4.6.1 for a long time because a lot of people have pinged me about supporting older installs of Windows OS, and so we want to continue to do that wherever possible. I could go on a rant here about licensing in training/exercise environments on Windows machines, but I'll spare everyone. I think the other systems in ghosts will stay on dotnet 6 for a long time, as it is a long term support version of dotnet.

3. You'll need to drop gecko/chrome driver in the bin folder for the browser actions to work, they are not distributed with this codebase.
